src/pybind/mgr/cephadm/migrations.py

```diff
@@ -259,6 +263,39 @@ def migrate_nfs_spec(self, service_id: str, pool: str, ns: Optional[str]) -> Non
                 self.mgr.log.warning(f'Failed to migrate export ({ret}): {err}\nExport was:\n{ex}')
         self.mgr.log.info(f'Done migrating nfs.{service_id}')
 
+    def migrate_3_4(self) -> bool:
+        registry_url = self.mgr.get_module_option('registry_url')
+        registry_username = self.mgr.get_module_option('registry_username')
+        registry_password = self.mgr.get_module_option('registry_password')
+        if registry_url and registry_username and registry_password:
+
+            registry_credentials = {'url': registry_url, 'username': registry_username, 'password': registry_password}
+            self.mgr.set_store('registry_credentials', json.dumps(registry_credentials))
+
+            self.mgr.set_module_option('registry_url', None)
+            self.mgr.check_mon_command({
+                'prefix': 'config rm',
+                'who': 'mgr',
+                'key': 'mgr/cephadm/registry_url',
+            })
```

Comment:
think this is redundant. See 

https://github.com/ceph/ceph/blob/8713c241a1bb1a6a41dea43e922e5124193fcf01/src/mgr/PyModule.cc#L199-L204

Reply:
without this the values do not get removed from config dump